examples/td3/td3.py

```diff
@@ -45,140 +47,156 @@ def main(cfg: "DictConfig"):  # noqa: F821
+    update_counter = 0
 
-    for i, tensordict in enumerate(collector):
+    sampling_start = time.time()
+    for tensordict in collector:
+        sampling_time = time.time() - sampling_start
         exploration_policy.step(tensordict.numel())
-        # update weights of the inference policy
+
+        # Update weights of the inference policy
         collector.update_policy_weights_()
 
-        if r0 is None:
-            r0 = tensordict["next", "reward"].sum(-1).mean().item()
         pbar.update(tensordict.numel())
 
         tensordict = tensordict.reshape(-1)
         current_frames = tensordict.numel()
+        # Add to replay buffer
         replay_buffer.extend(tensordict.cpu())
         collected_frames += current_frames
 
-        # optimization steps
+        # Optimization steps
+        training_start = time.time()
         if collected_frames >= init_random_frames:
             (
                 actor_losses,
                 q_losses,
             ) = ([], [])
-            for j in range(num_updates):
-                # sample from replay buffer
+            for _ in range(num_updates):
+                update_counter += 1
+                # Sample from replay buffer
                 sampled_tensordict = replay_buffer.sample().clone()
 
+                # Compute loss
                 loss_td = loss_module(sampled_tensordict)
 
                 actor_loss = loss_td["loss_actor"]
                 q_loss = loss_td["loss_qvalue"]
 
+                # Update critic
                 optimizer_critic.zero_grad()
-                update_actor = j % delayed_updates == 0
+                update_actor = update_counter % delayed_updates == 0
                 q_loss.backward(retain_graph=update_actor)
                 optimizer_critic.step()
                 q_losses.append(q_loss.item())
 
+                # Update actor
                 if update_actor:
                     optimizer_actor.zero_grad()
                     actor_loss.backward()
                     optimizer_actor.step()
+
                     actor_losses.append(actor_loss.item())
 
-                    # update qnet_target params
+                    # Update target params
                     target_net_updater.step()
 
-                # update priority
+                # Update priority
                 if prb:
                     replay_buffer.update_priority(sampled_tensordict)
 
-        rewards.append(
-            (i, tensordict["next", "reward"].sum().item() / env_per_collector)
-        )
-        train_log = {
-            "train_reward": rewards[-1][1],
-            "collected_frames": collected_frames,
-        }
-        if q_loss is not None:
-            train_log.update(
-                {
-                    "actor_loss": np.mean(actor_losses),
-                    "q_loss": np.mean(q_losses),
-                }
+        training_time = time.time() - training_start
```

Comment:
We should not need that (if we're using wandb or any other logger) but happy to keep it if that makes logging the time spent clearer

Reply:
I think it gives a better overview of how much time is spent at each big step (sample, update, and test) in the training loop. But I'm also okay with removing it as you said wandb and other logger keep track of timing themselves.

Reply:
we can keep that no big deal